src/object/background.cpp

```diff
@@ -356,15 +360,21 @@ Background::draw_image(DrawingContext& context, const Vector& pos_)
 
             if (m_image_top && (y < 0))
             {
-              canvas.draw_surface(m_image_top, p, 0.f, m_color, m_blend, m_layer);
+              m_image_top->set_color(m_color);
+              m_image_top->set_blend(m_blend);
+
+              m_image_top->draw(canvas, p, m_layer);
             }
             else if (m_image_bottom && (y > 0))
             {
-              canvas.draw_surface(m_image_bottom, p, 0.f, m_color, m_blend, m_layer);
+              m_image_bottom->set_color(m_color);
+              m_image_bottom->set_blend(m_blend);
+
+              m_image_bottom->draw(canvas, p, m_layer);
             }
             else
             {
-              canvas.draw_surface(m_image, p, 0.f, m_color, m_blend, m_layer);
+              m_image->draw(canvas, p, m_layer);
```

Comment:
Aren't you forgetting setting color and blend here?

Reply:
Lines 303, 304 have already set them.